lib/graphql/subscriptions/action_cable_subscriptions.rb

```diff
@@ -211,6 +212,11 @@ def stream_subscription_name(subscription_id)
       def stream_event_name(event)
         [EVENT_PREFIX, @transmit_ns, event.topic].join
       end
+
+      # cache message in case the same message is sent to events with different fingerprints
+      def decode_message(message)
+        @decoded_messages[message] ||= @serializer.load(message)
+      end
```

Comment:
I'm concerned about the lifetime of this cache, specifically, imagine: 

- An event is triggered with an object's gid in the payload: `trigger(:profile_was_updated, { user: User.find(1) }, ...)`
- That payload is passed to `decode_message`, which looks up `User.find(1)` and caches it for `"gid://User/1"`
- Later, the profile is updated again, with the same `trigger(...)` call above (so the same payload, too) 
- Won't  `decode_message(...)` return a stale `User` instance from the cache, instead of re-fetching it to get the latest changes? 

Or, is there some reason it wouldn't work out that way, and we could be sure that no stale data would be served from the cache?

Reply:
That seems to be a valid concern. 
I'm not even completely sure we need this cache. Did I understand correctly, that [this](https://github.com/rmosolgo/graphql-ruby/pull/3443/files#diff-a1ea5af1a78917fa653eff83d9ae72180fabd3d9cf616f14dd1871739963cd73R156-R158) code is gonna be run for every subscriber in case the subscription is not "broadcastable"?

Reply:
Yes, that's right -- if a subscription isn't "broadcastable", then it gets a unique fingerprint instead of a fingerprint based on the query and context. 

You might be able to use the lexical scope as a cache, for example, assigning `object = nil` above line 152, then using `object ||= ...` inside the loop. That way, you'd have a short-lived cache that lasted for the current update, but didn't spill into future updates.

Reply:
@rmosolgo updated accordingly. Please take a look